### DISCUSSION TITLE
Restructuring Project Directory for Better Organization

### DISCUSSION THREAD

**Discussion by only-issues** [wxt-dev/wxt#1161] [Q&A] [answered]
After reviewing the documentation, I noticed the current directory structure:

```
entrypoints/background.ts (only one background file allowed)
entrypoints/xyz.content/index.ts
entrypoints/xyz1.content/index.ts
... and many other content files
```

I have two main concerns:

1. **Single Background File Limitation**: 
   - Why is only one background file allowed?
   - Does this mean all background code must be written in a single file?

2. **Content File Organization**:
   - Why aren't content files grouped in a separate subdirectory for better organization?
   - Wouldn't a structure like this be cleaner?

   ```
   entrypoints/contents/xyz.content/index.ts
   entrypoints/contents/xyz1.content/index.ts
   ... and other content files
   ```

These changes could potentially improve code organization and maintainability. I'd appreciate your thoughts on these suggestions and any reasons for the current structure.

**Comment by aklinker1** [comment, marked as answer]
> Single Background File Limitation

You can import code from other files in your background, so no, it doesn't have to be a single file. You can also use a folder for the background and put related files in that folder, see https://wxt.dev/guide/essentials/entrypoints.html#background

```ts
// entrypoints/background/index.ts
import { initAlarms } from './alarms'
import { initMessaging } from './messaging'

export default define background(() => {
  initMessaging()
  initAlarms()
  // ...
})
```

```ts
// entrypoints/background/alarms.ts
export function initAlarms() {
  // ...
}
```

```ts
// entrypoints/background/messaging.ts
export function initMessaging() {
  // ...
}
```

Also see https://github.com/wxt-dev/wxt/issues/1038, someone has opened an issue for letting WXT discover multiple files rather than requiring you to import all the files you want to init.

> Content File Organization

There's been discussion around this in the past but no one has made a compelling enough argument for a different structure.

See https://github.com/wxt-dev/wxt/issues/192#issuecomment-1883435243
